### PR TITLE
[Vote 830] Disable cryo pods for 30 minutes

### PR DESCRIPTION
**Pull request by adamsong**
# Github documenting your Pull Request

As per [council vote 830](https://forums.yogstation.net/threads/vote-830-disable-the-ability-to-cryo-for-30-minutes-people-should-commit-to-30-mins-and-ahelp-if-they-cant-special-circumstances.23902/) disables the cryo pods from manual use for the first 30 minutes of the game. During this time attempting to cryo will produce an error telling the user the time remaining, and the cryo pods will be closed and not glowing. At the 30 minute mark, all the pods will open.

# Changelog

:cl:  
tweak: Disabled cryo pods for the first 30 minutes as per council vote 830
/:cl:

### CODE REVIEW

**Comment by Sniblet** [comment]
Cryopods are an alternative to ghosting, removing cryopods at any point just means people will ghost instead

**Comment by adamsong** [comment]
This PR was made to facilitate the passing of [council vote 830](https://forums.yogstation.net/threads/vote-830-disable-the-ability-to-cryo-for-30-minutes-people-should-commit-to-30-mins-and-ahelp-if-they-cant-special-circumstances.23902/). Responses to this criticizing the idea should instead be taken up in that thread.

**Comment by ynot01** [comment]
Would be appreciated if they were told the option of ahelping when attempting to use the pods

**Comment by adamsong** [comment]
How is this awaiting a passed council vote?

File: code/game/machinery/cryopod.dm [hunk 1]
```diff
@@ -155,12 +155,14 @@ GLOBAL_LIST_INIT(typecache_cryoitems, typecacheof(list(
 	updateUsrDialog()
 	return
 
+GLOBAL_VAR_INIT(cyropods_enabled, FALSE)
```
Comment (Bibby0110) [inline code/game/machinery/cryopod.dm]: cyropods

File: code/controllers/subsystem/ticker.dm [hunk 2]
```diff
@@ -143,6 +143,11 @@ SUBSYSTEM_DEF(ticker)
 	return ..()
 
 /datum/controller/subsystem/ticker/fire()
+	if(world.time > 30 MINUTES && !GLOB.cyropods_enabled)
```
Comment (Bibby0110) [inline code/controllers/subsystem/ticker.dm]: tEchNicAlLy world.time is from bootup so people will be able to cryo before 30 minutes round-time has passed

File: code/game/machinery/cryopod.dm [hunk 3]
```diff
@@ -355,6 +361,10 @@ GLOBAL_LIST_INIT(typecache_cryoitems, typecacheof(list(
 	if(!istype(target) || user.incapacitated() || !target.Adjacent(user) || !Adjacent(user) || !ismob(target) || (!ishuman(user) && !iscyborg(user)) || !istype(user.loc, /turf) || target.buckled)
 		return
 
+	if(!GLOB.cyropods_enabled)
+		to_chat(user, "<span class='boldnotice'>The cryogenics pod is currently disabled. It will be enabled in [round(((30 MINUTES) - world.time) / (1 MINUTES))] minutes</span>")
```
Comment (Bibby0110) [inline code/game/machinery/cryopod.dm]: technically called a freezer and not a pod

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/ticker.dm (hunk 2 above)
Comment (adamsong) [reply code/controllers/subsystem/ticker.dm]: Do we need another vote to determine if this 3 minutes is too big of a time lag?

Comment (adamsong) [reply code/controllers/subsystem/ticker.dm]: Council do not care

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**